src/ILLink.Shared/SharedStrings.resx

```diff
@@ -903,6 +903,36 @@
   <data name="DynamicallyAccessedMembersMismatchTypeArgumentTargetsFieldMessage" xml:space="preserve">
     <value>value stored in field '{0}' does not satisfy {3} requirements. The generic parameter '{1}' of '{2}' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.</value>
   </data>
+  <data name="DynamicallyAccessedMembersMismatchByRefParameterTargetsParameterTitle" xml:space="preserve">
+    <value>Target parameter argument does not satisfy 'DynamicallyAccessedMembersAttribute' in call to target method. The source by-reference parameter does not have matching annotations.</value>
+  </data>
+  <data name="DynamicallyAccessedMembersMismatchByRefParameterTargetsParameterMessage" xml:space="preserve">
+    <value>'{0}' argument does not satisfy {4} in call to '{1}'. The by-reference parameter '{2}' of method '{3}' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.</value>
+  </data>
+  <data name="DynamicallyAccessedMembersMismatchByRefParameterTargetsMethodReturnTypeTitle" xml:space="preserve">
+    <value>Target method return value does not satisfy 'DynamicallyAccessedMembersAttribute' requirements. The source by-reference parameter does not have matching annotations.</value>
+  </data>
+  <data name="DynamicallyAccessedMembersMismatchByRefParameterTargetsMethodReturnTypeMessage" xml:space="preserve">
+    <value>'{0}' method return value does not satisfy {3} requirements. The by-reference parameter '{1}' of method '{2}' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.</value>
+  </data>
+  <data name="DynamicallyAccessedMembersMismatchByRefParameterTargetsFieldTitle" xml:space="preserve">
+    <value>Value stored in target field does not satisfy 'DynamicallyAccessedMembersAttribute' requirements. The source by-reference parameter does not have matching annotations.</value>
+  </data>
+  <data name="DynamicallyAccessedMembersMismatchByRefParameterTargetsFieldMessage" xml:space="preserve">
+    <value>value stored in field '{0}' does not satisfy {3} requirements. The by-reference parameter '{1}' of method '{2}' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.</value>
```

Comment:
Yeah, I was debating that too, but I thought it would be important to differentiate the two warnings that would be provided here:

```C#
[DAM(DAMT.PublicMethods)]
Type _field;

void Method ([DAM(DAMT.PublicFields)] ref Type parameter)
{
}

static void Method2 ([DAM(DAMT.PublicFields)] Type parameter2)
{
  _field = parameter2; // Parameter targets field
  Method (ref _field); // By-reference parameter targets field
}
```

But if we think it makes more sense to use the same warnings and use ParameterValue instead of ByRefParameterValue I can remove the new ones.